packages/core/src/asciidoctor-core-api.js

```diff
@@ -1857,6 +1857,41 @@ Document.prototype.hasHeader = function () {
   return this['$header?']()
 }
 
+/**
+ * Replay attribute assignments at the block level
+ *
+ * @param {Object} blockAttributes - A JSON of attributes
+ *
+ * @memberof Document
+ */
+Document.prototype.playbackAttributes = function (blockAttributes) {
+  blockAttributes = toHash(blockAttributes)
+  if (blockAttributes) {
+    var attrEntries = blockAttributes['$[]']('attribute_entries')
+    if (attrEntries && Array.isArray(attrEntries)) {
+      var result = []
+      for (var i = 0; i < attrEntries.length; i++) {
+        var attrEntry = attrEntries[i]
+        var attrEntryObject = attrEntry
+        if (attrEntry && typeof attrEntry === 'object' && attrEntry.constructor.name === 'Object') {
+          attrEntryObject.$name = function () {
+            return this.name
+          }
+          attrEntryObject.$value = function () {
+            return this.value
+          }
+          attrEntryObject.$negate = function () {
+            return this.negate
+          }
+        }
```

Comment:
Does this essentially turn a json object into an Opal-compatible object?

Reply:
Yes, Asciidoctor (Ruby) will call the `attr_reader` on the `AttributeEntry` class: https://github.com/asciidoctor/asciidoctor/blob/88bc891d6a0143f3a2142f41d29cabe3cb21fe3f/lib/asciidoctor/document.rb#L91 so the object must respond to `$name`, `$value` and `$negate`.

In other words, I'm creating a very primitive `AttributeEntry` class.